src/main/scala/org/globalforestwatch/summarystats/gfwpro_dashboard/GfwProDashboardAnalysis.scala

```diff
@@ -23,32 +23,72 @@ object GfwProDashboardAnalysis extends SummaryAnalysis {
 
   val name = "gfwpro_dashboard"
 
+  /** Run the GFWPro dashboard analysis.
+    *
+    * If doGadmIntersect is true, read in the entire gadm feature dataset and
+    * intersect with the user feature list to determine the relevant gadm areas. If
+    * doGadmIntersect is false (usually for small number of user features), then
+    * determine the relevant gadm areas by using the raster gadm datasets GadmAdm0,
+    * GadmAdm1, and GadmAdm2.
+    */
   def apply(
     featureRDD: RDD[ValidatedLocation[Geometry]],
     featureType: String,
-    contextualFeatureType: String,
-    contextualFeatureUrl: NonEmptyList[String],
+    doGadmIntersect: Boolean,
+    gadmFeatureUrl: NonEmptyList[String],
     fireAlertRDD: SpatialRDD[Geometry],
     spark: SparkSession,
     kwargs: Map[String, Any]
   ): ValidatedWorkflow[Location[JobError],(FeatureId, GfwProDashboardData)] = {
     featureRDD.persist(StorageLevel.MEMORY_AND_DISK)
 
     val summaryRDD = ValidatedWorkflow(featureRDD).flatMap { rdd =>
-      val spatialContextualDF = SpatialFeatureDF(contextualFeatureUrl, contextualFeatureType, FeatureFilter.empty, "geom", spark)
-      val spatialContextualRDD = Adapter.toSpatialRdd(spatialContextualDF, "polyshape")
-      val spatialFeatureRDD = RDDAdapter.toSpatialRDDfromLocationRdd(rdd, spark)
+      val enrichedRDD = if (doGadmIntersect) {
+        println("Doing intersect with vector gadm")
+        val spatialContextualDF = SpatialFeatureDF(gadmFeatureUrl, "gadm", FeatureFilter.empty, "geom", spark)
+        val spatialContextualRDD = Adapter.toSpatialRdd(spatialContextualDF, "polyshape")
+        val spatialFeatureRDD = RDDAdapter.toSpatialRDDfromLocationRdd(rdd, spark)
 
-      /* Enrich the feature RDD by intersecting it with contextual features
-       * The resulting FeatureId carries combined identity of source feature and contextual geometry
-       */
-      val enrichedRDD =
+        /* Enrich the feature RDD by intersecting it with contextual features
+         * The resulting FeatureId carries combined identity of source feature and contextual geometry
+         */
         SpatialJoinRDD
           .flatSpatialJoin(spatialContextualRDD, spatialFeatureRDD, considerBoundaryIntersection = true, usingIndex = true)
           .rdd
           .flatMap { case (feature, context) =>
-            refineContextualIntersection(feature, context, contextualFeatureType)
+            refineContextualIntersection(feature, context, "gadm")
+          }
+      } else {
+        println("Using raster gadm")
+        rdd.map {
```

Comment:
I can't seem to find where this `rdd` is defined, is it the `featureRDD`?

Reply:
This rdd is defined at line 45 - it's the argument of the main flatMap call.  So, its the elements of featureRDD that are current valid (have not attached problem/error already).